zerver/lib/upload/s3.py

```diff
@@ -417,16 +421,44 @@ def get_export_tarball_url(self, realm: Realm, export_path: str) -> str:
         client = self.avatar_bucket.meta.client
         if export_path.startswith("/"):
             export_path = export_path.removeprefix("/")
-        signed_url = client.generate_presigned_url(
-            ClientMethod="get_object",
-            Params={
-                "Bucket": self.avatar_bucket.name,
-                "Key": export_path,
-            },
-            ExpiresIn=0,
-        )
-        # Strip off the signing query parameters, since this URL is public
-        return urlsplit(signed_url)._replace(query="").geturl()
+        if self.export_bucket:
+            if export_path.startswith("exports/"):
+                # Fix old data if the row was created when an export
+                # bucket was not in use:
+                export_path = export_path[len("exports/") :]
+            return client.generate_presigned_url(
+                ClientMethod="get_object",
+                Params={
+                    "Bucket": self.export_bucket.name,
+                    "Key": export_path,
+                },
+                # Expires in one week, the longest allowed by AWS
+                ExpiresIn=60 * 60 * 24 * 7,
+            )
```

Comment:
Is this intended to not do the same 
```
            # Strip off the signing query parameters, since this URL is public
            return urlsplit(signed_url)._replace(query="").geturl()
```
operation here as we do in the `else` block below ?

Reply:
Correct.  These URLs are actually time-bounded, which means that we can start expiring the uploads.  The exports which are shoved into the avatars bucket are protected only by security-through-obscurity.